rfd/0153-resource-guidelines.md

```diff
@@ -369,6 +374,82 @@ func (s *FooService) UpdateFoo(ctx context.Context, foo *foov1.Foo) (*foov1.Foo,
 }
 ```
 
+#### Patch
+
+The patch operation is intended for handling partial updates of a resource. The
+patch operation accepts the name of the resource and an update function. It 
+fetches the resource, calls the update function with the fetched resource and
+then calls `ConditionalUpdate` with the updated resource. It then returns the
+updated resource with the revision generated by the backend set.
+
+The revision of the fetched resource is used for the `ConditionalUpdate`. The
+Patch method should reject mutations that change the revision of the fetched
+resource. If a consumer wishes to provide the revision, they should instead
```

Comment:
We could let callers specify the base revision for the update, ie what the baseline resource was for that update, and if that changed refuse the update as well. In other words, allow for optimistic concurrency.

Reply:
See this thread https://github.com/gravitational/teleport/pull/38857#discussion_r1509256050

Reply:
Thanks, let's move the conversation there.